# adapted from lzma-4.42.0alphaX/extra/tar/file-4.21.patch
# 2008-10-05 lzma doesn't write this new format yet (as of
#   lzma 4.43 revision 2), but when it does, file will be ready
#   See discussion at 
#   http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=364260
#
--- magic/Magdir/compress.orig	2008-10-05 08:59:54.000000000 -0400
+++ magic/Magdir/compress	2008-10-05 09:08:57.000000000 -0400
@@ -189,6 +189,19 @@
 # bug #364260)
 #0	string		]\000\000\200\000	LZMA compressed data
 
+# LZMA (Lempel-Ziv-Markov chain-Algorithm) file format supported by:
+#   - 7-Zip <http://7-zip.org/>
+#   - LZMA SDK <http://7-zip.org/sdk.html>
+#   - LZMA Utils <http://tukaani.org/lzma/>
+#
+# Note that this is different format than the original format created
+# by LZMA_Alone from older versions of LZMA SDK. These files are
+# non-trivial to detect, because they have no magic bytes.
+#
+0	ustring		\xFFLZMA\x00	lzma compressed data,
+>6	ubyte&0x10	0x00		single-block stream
+>6	ubyte&0x10	0x10		multi-block stream
+
 # AFX compressed files (Wolfram Kleff)
 2	string		-afx-		AFX compressed file data

 
--- src/compress.c.orig	2007-03-05 03:41:29.000000000 +0100
+++ src/compress.c	2007-09-12 21:40:54.000000000 -0000
@@ -76,6 +76,7 @@
 	{ "PK\3\4",   4, { "gzip", "-cdq", NULL }, 1 },		/* pkzipped, */
 					    /* ...only first file examined */
 	{ "BZh",      3, { "bzip2", "-cd", NULL }, 1 },		/* bzip2-ed */
+	{ "\xFFLZMA\x00", 6, { "lzma", "-cd", NULL }, 1 },	/* new LZMA */
 };
 
 private size_t ncompr = sizeof(compr) / sizeof(compr[0]);
